Dockerfile

```diff
@@ -60,9 +61,42 @@ RUN apk add -u shadow libc6-compat && \
 
 USER secrets-provider
 
+# =================== DEBUG BUILD LAYER ===================
+# this layer is used to build the debug binaries
+FROM secrets-provider-builder-base as secrets-provider-builder-debug
+
+# Build Delve
+RUN go get github.com/go-delve/delve/cmd/dlv
+
+EXPOSE 40000
+
+COPY . .
```

Comment:
@abrahamko this Dockerfile is quite hard to follow. can we have all the builders stacked in the top of the file? 

i think a better order will be:
- secrets-provider-builder-base
- secrets-provider-builder
- secrets-provider-builder-debug
- busybox
- secrets-provider-base
- secrets-provider
- secrets-provider-debug
- secrets-provider-for-k8s-redhat

Reply:
also - it's not from your PR but can we have an alignment in the naming and choose between `secrets-provider` and `secrets-provider-for-k8s`? either is fine by me.

Reply:
@orenbm I like the order you put here.
Just so you know, there's a reason for the order I did because using `--target` flag for docker build invokes ALL steps until the specified target.
In our case, it means building for debug builds for also for release EVERY TIME.
There is a way to handle it, though, as there's an addition of a BuildKit to docker that executes only the needed steps for the specified target, but it is not used by default.
Would you like me to add this to the script?

Reply:
i think that readability is more important in this case. if the `BuildKit` addition is not to big then go for it. If not, i still think that it's fine to build redundant parts (as long as they don't take too long) to improve the readability of such a large Dockerfile.

Reply:
wdyt?

Reply:
I added the buildkit. 😉